include/linux/device.h

```diff
@@ -25,4 +25,13 @@ class_find_device_by_of_node(struct class *class, const struct device_node *np)
 }
 #endif /* < KERNEL_VERSION(5, 4, 0) */
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 253) || \
+(LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(5, 9, 0))) && RHEL_RELEASE_CODE < 0x804
+static inline int dev_err_probe(const struct device *dev, int err, const char *fmt, ...)
+{
+	dev_err(dev, fmt);
```

Comment:
Apologies for missing this in the first round, but this does not forward the variable arguments. How about this simplified variant of the [upstream function](https://elixir.bootlin.com/linux/latest/source/drivers/base/core.c#L4955), without `device_set_deferred_probe_reason()` which is absent on the CentOS kernels?
```suggestion
	struct va_format vaf;
	va_list args;

	va_start(args, fmt);
	vaf.fmt = fmt;
	vaf.va = &args;
	dev_err(dev, "error %pe: %pV", ERR_PTR(err), &vaf);
	va_end(args);
```

Reply:
Thanks, yes that's the way to handle variable arguments. I didn't pay enough attention. It is good to keep it close to the upstream function. I will update the commit with this change.

Reply:
I can actually use the original function from commit a787e5400a1c, that also handles -EPROBE_DEFER.